index.js

```diff
@@ -15,6 +15,15 @@ module.exports = function (res) {
 		unzip.statusMessage = res.statusMessage;
 		unzip.socket = res.socket;
 
+		unzip.once('error', function (err) {
+			if (err.code === 'Z_BUF_ERROR') {
+				res.emit('end');
+				return;
+			}
+
+			res.emit('error', err);
```

Comment:
Seems like it doesn't exist on `res`: `TypeError: res.destroy is not a function`.